src/libs/actions/IOU.js

```diff
@@ -722,11 +723,326 @@ function payIOUReport({
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport.reportID}`,
+            value: {
+                [optimsticReportActionID.sequenceNumber]: {
+                    pendingAction: null,
+                },
+            },
+        },
+    ];
+
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport.reportID}`,
+            value: {
+                [optimsticReportActionID.sequenceNumber]: {
+                    ...optimsticReportActionID,
+                    errors: {
+                        [DateUtils.getMicroseconds()]: Localize.translateLocal('iou.error.genericCreateFailureMessage'),
+                    },
+                },
+            },
+        },
+    ];
+
+    // Now, let's add the data we need just when we are creating a new chat report
+    if (isNewChat) {
+        const optimisticCreateAction = ReportUtils.buildOptimisticCreatedReportAction(recipientEmail);
+
+        // Change the method to set for new reports because it doesn't exist yet, is faster,
+        // and we need the data to be available when we navigate to the chat page
+        optimisticData[0].onyxMethod = CONST.ONYX.METHOD.SET;
+        optimisticData[0].value = {
+            ...optimisticData[0].value,
+            pendingFields: {createChat: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD},
+        };
+        optimisticData[1].onyxMethod = CONST.ONYX.METHOD.SET;
+        optimisticData[1].value = {
+            ...optimisticCreateAction,
+            ...optimisticData[1].value,
+        };
+        optimisticData[2].onyxMethod = CONST.ONYX.METHOD.SET;
+    }
+
+    return {
+        params: {
+            iouReportID: optimisticIOUReport.reportID,
+            chatReportID: chatReport.reportID,
+            reportActionID: optimsticReportActionID.reportActionID,
+            paymentMethodType,
+            transactionID: optimsticReportActionID.originalMessage.IOUTransactionID,
+            clientID: optimsticReportActionID.sequenceNumber,
+            newIOUReportDetails,
+        },
+        optimisticData,
+        successData,
+        failureData,
+    };
+}
+
+/**
+ * @param {Object} chatReport
+ * @param {Object} iouReport
+ * @param {Object} recipient
+ * @param {String} paymentMethodType
+ * @returns {Object}
+ */
+function getPayMoneyRequestParams(chatReport, iouReport, recipient, paymentMethodType) {
+    const newSequenceNumber = Report.getMaxSequenceNumber(chatReport.reportID) + 1;
+
+    const optimsticReportActionID = ReportUtils.buildOptimisticIOUReportAction(
+        newSequenceNumber,
+        CONST.IOU.REPORT_ACTION_TYPE.PAY,
+        iouReport.total,
+        iouReport.currency,
+        '',
+        [recipient],
+        paymentMethodType,
+        '',
+        iouReport.reportID,
+    );
+
+    // First, add data that will be used in all cases
```

Comment:
This is the only case in this method, since we don't have to create a `chatReport`